apps/onnx/model.cpp

```diff
@@ -2,483 +2,573 @@
 #include <pybind11/pybind11.h>
 #include <pybind11/stl.h>
 
-#include "common_types.h"
-#include "onnx_converter.h"
+#include <sys/time.h>
 #include <fstream>
 #include <random>
-#include <sys/time.h>
 #include <unordered_set>
+#include "common_types.h"
+#include "denormal_disabler.h"
+#include "benchmarking_utils.h"
+#include "onnx_converter.h"
 
 namespace py = pybind11;
 
 namespace {
 
 HalideModel convert_onnx_model(
-    const std::string &onnx_model_str) {
-    onnx::ModelProto onnx_model;
-    onnx_model.ParseFromString(onnx_model_str);
-
-    if (onnx_model.graph().output_size() == 0) {
-        throw std::invalid_argument("No output specified in the model");
-    }
-
-    std::unordered_set<std::string> dflt_values;
-    for (const auto &dflt : onnx_model.graph().initializer()) {
-        dflt_values.insert(dflt.name());
-    }
-    HalideModel result;
-    for (const auto &input : onnx_model.graph().input()) {
-        if (dflt_values.find(input.name()) != dflt_values.end()) {
-            continue;
-        }
-        result.input_names.push_back(input.name());
-        result.input_types[input.name()] = input.type().tensor_type().elem_type();
-    }
-    for (const auto &output : onnx_model.graph().output()) {
-        result.output_names.push_back(output.name());
-        result.output_types.push_back(output.type().tensor_type().elem_type());
-    }
-
-    result.model.reset(new Model(convert_model(onnx_model)));
-
-    std::vector<Halide::Func> funcs;
-    for (const auto &output : onnx_model.graph().output()) {
-        const auto &tensor = result.model->outputs.at(output.name());
-        funcs.push_back(tensor.rep);
-    }
-    result.rep.reset(new Halide::Pipeline(funcs));
-
-    for (const Halide::Expr &requirement : result.model->requirements) {
-        if (Halide::Internal::is_pure(requirement)) {
-            result.rep->add_requirement(requirement);
-        }
-    }
-
-    return result;
+    const std::string& onnx_model_str,
+    const std::unordered_map<std::string, int>& expected_dim_sizes,
+    const IOLayout layout) {
+  onnx::ModelProto onnx_model;
+  onnx_model.ParseFromString(onnx_model_str);
+
+  if (onnx_model.graph().output_size() == 0) {
+    throw std::invalid_argument("No output specified in the model");
+  }
+
+  std::unordered_set<std::string> dflt_values;
+  for (const auto& dflt : onnx_model.graph().initializer()) {
+    dflt_values.insert(dflt.name());
+  }
+  HalideModel result;
+  for (const auto& input : onnx_model.graph().input()) {
```

Comment:
here and many other places: `type &name` is preferred style in Halide (rather than `type& name`)

Reply:
I forgot to apply clang-format to this file. This is fixed now.